src/stactools/noaa_nclimgrid/utils.py

```diff
@@ -48,6 +48,26 @@ def data_frequency(href: str) -> Frequency:
     return frequency
 
 
+def collection_type(href: str) -> CollectionType:
+    """Determine if data belongs to a 'monthly', 'daily-prelim', or 'daily-scaled'
+    collection.
+
+    Args:
+        href (str): HREF to a NClimGrid netCDF or COG file.
+
+    Returns:
+        CollectionType: Enum of 'monthly', 'daily-prelim', or 'daily-scaled'.
+    """
+    basename = os.path.splitext(os.path.basename(href))[0]
+    if basename.startswith("nclimgrid"):
+        type = CollectionType.MONTHLY
+    elif "prelim" in basename:
+        type = CollectionType.DAILY_PRELIM
+    else:
+        type = CollectionType.DAILY_SCALED
+    return type
```

Comment:
IMO this makes more sense as a staticmethod on `CollectionType`, e.g.

```python
class CollectionType(Enum):
    @staticmethod
    def from_href(href: str) -> "CollectionType":
        ....
```

Reply:
Yazza. I like.